ui/app/store/actions.js

```diff
@@ -645,6 +645,95 @@ export function signPersonalMsg (msgData) {
   }
 }
 
+export function decryptMsgInline (msgData) {
+  log.debug('action - decryptMsgInline')
+  return (dispatch) => {
+    window.onbeforeunload = null
+    return new Promise((resolve, reject) => {
+      log.debug(`actions calling background.decryptMessageInline`)
+      background.decryptMessageInline(msgData, (err, newState) => {
+        log.debug('decryptMsgInline called back')
+        if (newState) {
+          msgData = newState.unapprovedDecryptMsgs[msgData.metamaskId]
+          dispatch(updateMetamaskState(newState))
+        } else {
+          log.error('Wrong data for decryptMessageInline')
```

Comment:
We need to have this for a case when the message for decryption is empty.

Reply:
I'm not sure I follow - what happens when the message for decryption is empty? I would expect it to either throw an error in that case, or return a full state object regardless. I can't see a path where it'd something falsey in place of this state object _and_ not throw an error.

Reply:
I've just tested it, and this has the effect of catching _all errors_ and returning a "Wrong data" message, rather than the error itself. We should be rejecting the actual `Error` object when an error occurs.